src/qibolab/result.py

```diff
@@ -1,53 +1,44 @@
 from dataclasses import dataclass
 from functools import cached_property
+from typing import Optional
 
 import numpy as np
 import numpy.typing as npt
-
-ExecRes = np.dtype([("i", np.float64), ("q", np.float64), ("shots", np.uint32)])
+from scipy import signal
 
 
 @dataclass
 class ExecutionResults:
     """Data structure to deal with the output of execute_pulse_sequence"""
 
-    array: npt.NDArray[ExecRes]
+    i: npt.NDArray[np.float64]
+    q: npt.NDArray[np.float64]
+    shots: Optional[npt.NDArray[np.uint32]] = None
 
     @classmethod
-    def from_components(cls, is_, qs, shots):
-        ar = np.empty(len(is_), dtype=ExecRes)
-        ar["i"] = is_
-        ar["q"] = qs
-        ar["shots"] = shots
-        ar = np.rec.array(ar)
-        return cls(ar)
-
-    @property
-    def i(self):
-        return self.array.i
-
-    @property
-    def q(self):
-        return self.array.q
+    def from_components(cls, is_, qs_, shots=None):
+        return cls(is_, qs_, shots)
 
     @property
-    def shots(self):
-        return self.array.shots
+    def in_progress(self):
+        """Placeholder for when we implement live fetching of data from instruments."""
+        return False
 
     @cached_property
     def msr(self):
         """Computes msr value."""
```

Comment:
Maybe we should avoid the acronym in this docstring.

Reply:
@aorgazf @DavidSarlle do you have a reference of what MSR actually means in this context so that we can document it properly? After a quick search in google as well as the qua and qblox docs I could not find any reference to the term.

Reply:
@stavros11  MSR means mesurement and it is equal to:  
**MSR[V]**: Resonator signal voltage mesurement in volts

Reply:
I agree we should spell out `measurement` at least in the docstring, but maybe even better if we can call `measurement` also the property itself.
It's not too many characters, we all have autocompletion, and it makes everything easier to understand.

Concerning `MSR[V]` I do not have a strong opinion at the moment, it can also stay as it is.

Reply:
I fully agree.

Reply:
I renamed `msr` to `measurement` and updated the docstring. Let me know if you agree, particularly because this is part of the interface so changing its properties may break various things, including qibocal. For now I checked and everything looks fine but generally we should keep this in mind when changing the result object.